### DISCUSSION TITLE
Lazyfetch array with objects instead of urls

### DISCUSSION THREAD

**Discussion by StavrosNicolaou** [Touffy/client-zip#65] [Q&A] [answered]
I've been using client-zip on deno to zip-on-the-fly some files on my origin server using the following code:

```
async function *lazyFetch() {
      for (const url of urls) yeild await fetch(url)
}
```

However with this way I cannot put files in different folders.

I would like to be able to have an array with an object for each file, for example:

```
let files = [
    {
        name: "folder1/file1.txt",
        size: 123,
        input: "https://file-download-url.com/file"
    }
]
```

and lazyfetch it. Is this doable?

Thank you.

**Comment by Touffy** [comment, marked as answer]
Absolutely. We get a Response as usual, and instead of yielding it by itself, we put it in an object with its desired filename:
```js
async function *lazyFetch() {
  for (const { name, input } of files) yield { name, input: await fetch(input) }
}
```

I didn't copy the 'size' property because it doesn't do anything when actually making the Zip. You can pass your original `files` array in the `metadata` option, so that `downloadZip` will set the correct Content-Length header (assuming your sizes are accurate, of course) :

```js
const zipResponse = downloadZip(lazyFetch(), { metadata: files })
```
